dominode-bootstrapper/dominode_bootstrapper/geonodeadmin.py

```diff
@@ -321,15 +400,35 @@ def _bootstrap_department_in_geoserver(
     existing_workspaces = manager.list_workspaces()
     workspace_name = department.value
     workspace_exists = False
-    # TODO: check if the workspace that we want to create already exists and do not continue if it does
+
     # 1. create the workspace
 
+    workspace = manager.get_workspace(workspace_name)
+
+    if workspace is None:
+        manager.create_workspace(workspace_name)
+    else:
+        typer.echo(f'Workspace {workspace_name} already exists, skipping...')
+        workspace_exists = True
+
     if not workspace_exists:
         # 2. create the geofence admin rules
         existing_rules = manager.list_geofence_admin_rules()
         group_name = get_geoserver_group_name(department)
         role_name = f'ROLE_{group_name}'
+
+        typer.echo(f'Existing rules {existing_rules}, role name {role_name}')
+
         if role_name not in [i['roleName'] for i in existing_rules]:
+            typer.echo(f'Creating Geoserver admin rule for {department.value}...')
             manager.create_geofence_admin_rule(
                 department.value, role_name, UserRole.EDITOR)
         # 3. create the postgis db store
+        # manager.create_postgis_store(
```

Comment:
@ricardogsilva should we used here db credentials from https://github.com/dominodeorg/dominode/blob/7c3e9fe207e8b2b77e2ae1b9e1dd384327dc884b/dominode-bootstrapper/dominode_bootstrapper/dominodeadmin.py#L20?

Reply:
No. The plan is to use credentials for a less privileged user, that only has read access to the `public` schema.

Assume that the user creds will be supplied to that function (and to the callers). I have another PR which deals with the creation of these additional DB users and brings in a way to specify their credentials via env or config files. But for the purpose of this PR, assume the credentials are going to be supplied by the user